packages/libs/lambda-at-edge/src/default-handler.ts

```diff
@@ -221,23 +222,42 @@ const handleOriginRequest = async ({
     s3Origin.path = `${basePath}/static-pages`;
     request.uri = pagePath.replace("pages", "");
     addS3HostHeader(request, normalisedS3DomainName);
+
     return request;
   }
 
   const tBeforePageRequire = now();
-  const page = require(`./${pagePath}`); // eslint-disable-line
+  let page = require(`./${pagePath}`); // eslint-disable-line
   const tAfterPageRequire = now();
 
   log("require JS execution time", tBeforePageRequire, tAfterPageRequire);
 
   const tBeforeSSR = now();
   const { req, res, responsePromise } = lambdaAtEdgeCompat(event.Records[0].cf);
-  page.render(req, res);
+  try {
+    // If page is _error.js, set status to 404 so _error.js will render a 404 page
+    if (pagePath === "pages/_error.js") {
+      res.statusCode = 404;
+    }
+
+    // Render page
+    await page.render(req, res);
+  } catch (error) {
+    // Set status to 500 so _error.js will render a 500 page
+    console.error(
+      `Error rendering page: ${pagePath}. Error:\n${error}\nRendering Next.js error page.`
+    );
+    res.statusCode = 500;
```

Comment:
Not completely sure why this works, I think the params might be slightly named incorrectly in case of _error.js page? Will need to look into the code more for the render() function in _error.js.

Reply:
Lgtm. I hadn't realised before that Next.js stopped serving 500 error page on `page.render` uncaught errors. https://github.com/vercel/next.js/commit/15cdb4f4089713b7f1e3079522b15bdd424ac27c

Reply:
Thanks for merging! Yeah, it looks like it was changed in 9.5? Might be worth to add some end-to-end tests on CloudFront in the future to verify with the latest Next.js versions (e.g 9.4.x and 9.5.x) as the unit and integration tests won't always catch these kinds of issues :)